djangochannelsrestframework/mixins.py

```diff
@@ -1,7 +1,56 @@
+from typing import Dict, Optional, OrderedDict, Union
+from rest_framework.utils.serializer_helpers import ReturnDict, ReturnList
+from django.db.models import Model, QuerySet 
 from rest_framework import status
 
 from .decorators import action
+from djangochannelsrestframework.settings import api_settings
+
+
+class PaginatedMixin:
+    
+    pagination_class = api_settings.DEFAULT_PAGINATION_CLASS
+
+    @property
+    def paginator(self) -> Optional[any]:
+        """Gets the paginator class
+
+        Returns:
+            Pagination class. Optional.
+        """
+        if not hasattr(self, "_paginator"):
+            if self.pagination_class is None:
+                self._paginator = None
+            else:
+                self._paginator = self.pagination_class()
+        return self._paginator
+
+    def paginate_queryset(
+        self, queryset: QuerySet[Model], **kwargs: Dict
+    ) -> Optional[QuerySet]:
+        if self.paginator is None:
+            return None
+        return self.paginator.paginate_queryset(
+            queryset, self.scope, view=self, **kwargs
+        )
 
+    def get_paginated_response(
+        self, data: Union[ReturnDict, ReturnList]
+    ) -> OrderedDict:
+        assert self.paginator is not None
+        return self.paginator.get_paginated_response(data)
+
+class StreamedPaginatedListMixin(PaginatedMixin):
+    
+    async def handle_action(self, action: str, request_id: str, **kwargs):
+        await super().handle_action(action, request_id, **kwargs)
+        while self.paginator.offset < self.paginator.count:
```

Comment:
not sure about this being by overriding `handle action` if you have other `@action` methods on your consumer this will get called...

Reply:
Ok, I will check another path

Reply:
I can't seem to find another path, my solution is to check the action to `list`